integration-tests/api/__tests__/admin/order-edit.js

```diff
@@ -766,6 +766,65 @@ describe("[MEDUSA_FF_ORDER_EDITING] /admin/order-edits", () => {
       )
     })
 
+    it("creates an order edit with custom creator", async () => {
+      const api = useApi()
+
+      const response = await api.post(
+        `/admin/order-edits/`,
+        {
+          order_id: orderId,
+          created_by: "custom_service",
```

Comment:
thought: I don't think we should allow this to be overridden at the API level.

Reply:
I can't see why we shouldn't expose it. Any particular reason?

Reply:
I am fine with keeping it out for now, as our authentication mechanism ensure that a user is always present on `/admin/*` endpoints, but was just curious.

Reply:
The reason is that it is odd that you can "impersonate" someone else despite being logged in and identified. It would likely be something that we would eventually have to remove when focusing more on audit logging for enterprise purposes.

I understand that it would be helpful to control this if a service is directly calling the endpoints of a Medusa instance. But in a scenario like this, the correct approach would be that the Service (e.g., MyOMSService) itself is authorized, and `created_by` would then automatically be populated with details about the service authorization (e.g., which API key was it that created this order edit). 

Since we don't have service-to-service authorization or API keys at this point I would rather have use push people to do custom endpoints to handle this instead of creating a quick trick that we would need to go back on later.

Wdyt?

Reply:
I have one suggestion, maybe we can have both the loggedInUser and the custom created by. something like `created_by` and `on_behalf_of` or something like that? so that we have the info of who makes the action and for who?

Reply:
> Since we don't have service-to-service authorization or API keys at this point, I would rather push people to do custom endpoints to handle this instead of creating a quick trick that we would need to go back on later.

Makes sense! I agree, this is the right approach for now. 

> I have one suggestion, maybe we can have both the loggedInUser and the custom created by. something like created_by and on_behalf_of or something like that

That's definitely a viable option, but I believe it's premature optimization (like my first take 😅) for this case, so I'd rather leave that for a separate iteration. Or until someone requests it. Cool with that?